procedures/automation-android-setup.md

```diff
@@ -0,0 +1,200 @@
+# How to set up automation for nightly Android product
+
+These instructions define which resources are needed for setting up nightly Android product release automation, as well as the steps to prepare the automation itself. Additionally, this guide include steps for running a "staging" version of the automation so that it can be tested in the developer's own repository.
+
+Throughout this document, wherever the term `$product` is used, substitute your product's name in (replacing spaces with hyphens), e.g.: `reference-browser` or `fenix`.
+
+Note: we don't need to explicitly define scopes in Taskcluster. We just define them in the task, hook's role, and each scriptworker script. Taskcluster will implicitly detect scopes, but it doesn't use them for anything other than generic equality/wildcard verification.
+
+1. Request dep and release signing keys (example bug for [`reference-browser`](https://bugzilla.mozilla.org/show_bug.cgi?id=1508761)). Confirm with the app's team what the "signature algorith" and "digest algorithm" should be, and include that information in the ticket.
+2. Clone the product's repository
+    * Note that, while doing "staging" runs, you'll have to use your `master` branch in your fork. Hook cloning/CoT/etc haven't been tested with a non-default branch
```

Comment:
Hmm, I'm not sure how the config and code I point to in the rest of the guide work with that - I might be handling a clone/commit the wrong way, which could :b:ork it.
I'll change this accordingly to just recommend using the `master` branch.
Maybe after taskgraph-ing, the in-repository logic will be simple enough that I can more-confidently recommend using a non-master branch!

Reply:
Ah yeah, I confirm this doesn't work as is on Fenix. I just encountered this error https://tools.taskcluster.net/groups/c_FN0u-WTBuqQoOryKBXnQ/tasks/c_FN0u-WTBuqQoOryKBXnQ/runs/0/logs/public%2Flogs%2Flive_backing.log#L437 because .taskcluster.yml re-clones the forked repo. I'll see if I can make this better. I need it to test https://github.com/mozilla-mobile/fenix/pull/397 out

Reply:
Great news! The effort to allow branches to work is minimal: https://github.com/mozilla-mobile/fenix/pull/397/commits/55b419cfcab926ddc73add838db05be65da1e3a1

Bonus points: we can reuse the repo baked in the docker image. This saves some network and CPU cycles \o/

Reply:
Wow, I didn't know that you wouldn't have to re-clone like that! That's awesome! :tada: :tada: